resilience4j-ratpack/src/main/java/io/github/resilience4j/ratpack/circuitbreaker/CircuitBreakerTransformer.java

```diff
@@ -72,16 +87,10 @@ public void success(T value) {
                     @Override
                     public void error(Throwable throwable) {
                         long durationInNanos = System.nanoTime() - start;
-                        circuitBreaker.onError(durationInNanos, throwable);
-                        try {
-                            if (recoverer != null) {
-                                down.success(recoverer.apply(throwable));
-                            } else {
-                                down.error(throwable);
-                            }
-                        } catch (Throwable t) {
-                            down.error(t);
+                        if (recordFailurePredicate.test(throwable)) {
+                            circuitBreaker.onError(durationInNanos, throwable);
                         }
```

Comment:
in the else, shouldn't we have circuitBreaker.onSuccess(durationInNanos) call?

Reply:
@amityatagiri @RobWin This appears to be redundant - as the same check is present in `CircuitBreakerStateMachine.onError`

Reply:
@amityatagiri @drmaas This is the case when a decorated service throws a business exception (not a technical exception) and the exceptions should be ignored and not count as an error or success.
The CircuitBreakerStateMachine emits a `CircuitBreakerOnIgnoredErrorEvent` in that case.

From my point of view the `CircuitBreakerTransformer` can invoke `circuitBreaker.onError` directly and must not do the check `circuitBreakerConfig.getRecordFailurePredicate().test(throwable)`, because this is implemented in `circuitBreaker.onError` anyway.